File: fetch_tx.js
const https = require('https');
const fs = require('fs');

const rpcs = [
  "polygon.meowrpc.com",
  "polygon-mainnet.public.blastapi.io",
  "1rpc.io",
  "rpc-mainnet.maticvigil.com",
  "rpc.polygon.bor.marlin.org"
];

const paths = [
  "/", "/", "/matic", "/", "/"
];

function fetchRPC(hostname, path, method, params) {
  const data = JSON.stringify({ jsonrpc: "2.0", method: method, params: params, id: 1 });
  const options = {
    hostname: hostname,
    port: 443,
    path: path,
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'Content-Length': data.length
    }
  };

  return new Promise((resolve, reject) => {
    const req = https.request(options, (res) => {
      let body = '';
      res.on('data', (d) => { body += d; });
      res.on('end', () => {
        try {
          const parsed = JSON.parse(body);
          if (parsed.error) {
            resolve(null); // error from rpc
          } else {
            resolve(parsed);
          }
        } catch (e) {
          resolve(null);
        }
      });
    });
    req.on('error', () => resolve(null));
    req.write(data);
    req.end();
  });
}

async function run() {
  let receipt = null;
  let txinfo = null;

  for (let i = 0; i < rpcs.length; i++) {
    console.log("Trying", rpcs[i]);
    receipt = await fetchRPC(rpcs[i], paths[i], "eth_getTransactionReceipt", ["0xc6792d297606cd299cb1e6c15b63c3e211e21a4ad60a6767cf8541c73d5d73a7"]);
    if (receipt && receipt.result) {
      txinfo = await fetchRPC(rpcs[i], paths[i], "eth_getTransactionByHash", ["0xc6792d297606cd299cb1e6c15b63c3e211e21a4ad60a6767cf8541c73d5d73a7"]);
      if (txinfo && txinfo.result) {
        console.log("Success with", rpcs[i]);
        fs.writeFileSync('tx_receipt.json', JSON.stringify(receipt, null, 2));
        fs.writeFileSync('tx_info.json', JSON.stringify(txinfo, null, 2));
        return;
      }
    }
  }
  console.log("All failed");
}
run();

File: fetch_tx.py
import requests
import json
import sys

rpcs = [
    "https://polygon-rpc.com/",
    "https://rpc-mainnet.maticvigil.com/",
    "https://polygon.meowrpc.com",
    "https://1rpc.io/matic",
    "https://polygon-mainnet.public.blastapi.io",
    "https://rpc.ankr.com/polygon"
]

tx_hash = "0xc6792d297606cd299cb1e6c15b63c3e211e21a4ad60a6767cf8541c73d5d73a7"
payload_receipt = {"jsonrpc": "2.0", "method": "eth_getTransactionReceipt", "params": [tx_hash], "id": 1}
payload_info = {"jsonrpc": "2.0", "method": "eth_getTransactionByHash", "params": [tx_hash], "id": 1}
headers = {'Content-Type': 'application/json'}

def fetch(payload):
    for rpc in rpcs:
        print(f"Trying {rpc}")
        try:
            r = requests.post(rpc, json=payload, headers=headers, timeout=5)
            if r.status_code == 200:
                data = r.json()
                if "error" not in data:
                    return data
                else:
                    print(f"Error from {rpc}: {data['error']}")
            else:
                print(f"HTTP {r.status_code} from {rpc}")
        except Exception as e:
            print(f"Exception from {rpc}: {e}")
    return None

print("Fetching receipt...")
receipt = fetch(payload_receipt)
if receipt:
    with open("tx_receipt.json", "w") as f:
        json.dump(receipt, f, indent=2)
    print("Saved tx_receipt.json")

print("Fetching info...")
info = fetch(payload_info)
if info:
    with open("tx_info.json", "w") as f:
        json.dump(info, f, indent=2)
    print("Saved tx_info.json")
